tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -164,78 +170,140 @@ def _check_ssm_success(ssm_client, command_id, instance_id):
+            nlb_stack_name = generate_stack_name(
+                f"integ-tests-MultiUserInfraStack{directory_type}NLB", request.config.getoption("stackname_suffix")
+            )
+            logging.info("Creating stack %s", nlb_stack_name)
+            # TODO: don't hardcode this ARN
+            certificate_arn = "arn:aws:acm:us-east-1:298836457240:certificate/102d3350-0311-47fd-acf3-ee1f1381725c"
+            with open(nlb_stack_template_path) as nlb_stack_template:
+                nlb_stack = CfnStack(
+                    name=nlb_stack_name,
+                    region=region,
+                    template=nlb_stack_template.read(),
+                    parameters=[
+                        {
+                            "ParameterKey": "LDAPSCertificateARN",
+                            "ParameterValue": certificate_arn,
+                        },
+                        {
+                            "ParameterKey": "VPCId",
+                            "ParameterValue": directory_stack.cfn_outputs["VpcId"],
+                        },
+                        {
+                            "ParameterKey": "SubnetId1",
+                            "ParameterValue": directory_stack.cfn_outputs["PrivateSubnetIds"].split(",")[0],
+                        },
+                        {
+                            "ParameterKey": "SubnetId2",
+                            "ParameterValue": directory_stack.cfn_outputs["PrivateSubnetIds"].split(",")[1],
+                        },
+                        {
+                            "ParameterKey": "SimpleADPriIP",
+                            "ParameterValue": directory_stack.cfn_outputs["DirectoryDnsIpAddresses"].split(",")[0],
+                        },
+                        {
+                            "ParameterKey": "SimpleADSecIP",
+                            "ParameterValue": directory_stack.cfn_outputs["DirectoryDnsIpAddresses"].split(",")[1],
+                        },
+                    ],
+                )
+            cfn_stacks_factory.create_stack(nlb_stack)
+            logging.info("Creation of NLB stack %s complete", nlb_stack_name)
+            created_directory_stacks[region]["nlb"] = nlb_stack_name
+        return directory_stack_name, nlb_stack_name
+
+    yield _directory_factory
+
+    for region, stack_dict in created_directory_stacks.items():
+        for stack_type, stack_name in stack_dict.items():
+            if request.config.getoption("no_delete"):
+                logging.info(
+                    "Not deleting %s stack named %s in region %s because --no-delete option was specified",
+                    stack_type,
+                    stack_name,
+                    region,
+                )
+            else:
+                logging.info("Deleting %s stack named %s in region %s", stack_type, stack_name, region)
+                boto3.client("cloudformation", region_name=region).delete_stack(StackName=stack_name)
```

Comment:
Sorry, this is a bug from me. It should've been
```
cfn_stacks_factory.delete_stack(stack_name, region)
```

Reply:
fixed.